test/datalog_tests/function.fail.ast.expected

```diff
@@ -15,3 +15,6 @@ error: ./test/datalog_tests/function.fail.dl:5:1-9:1: Recursive function definit
 error: ./test/datalog_tests/function.fail.dl:9:5-9:17: Type constructor in the left-hand side of an assignment is only allowed for types with one constructor,  but "option_t" has multiple constructors
 
 error: ./test/datalog_tests/function.fail.dl:11:16-11:27: Type constructor in the left-hand side of an assignment is only allowed for types with one constructor,  but "option_t" has multiple constructors
+
+error: ./test/datalog_tests/function.fail.dl:6:5-7:1: Expression parameterized(string, int) has unknown type in CtxTop
```

Comment:
This error message looks confusing. Shouldn't the error indicate that unification failed between the types of x and y?

Reply:
You're right. This is a result of how type inference works right now. I think I know how to fix this. I will add an issue for the moment and will fix this in a separate PR.